velox/vector/FlatVector.h

```diff
@@ -394,8 +394,15 @@ class FlatVector final : public SimpleVector<T> {
     return true;
   }
 
+  // Acquire ownership for any string buffer that appears in source, the
+  // function does nothing if the vector type is not Varchar or Varbinary.
+  // The function throws if input encoding is lazy.
   void acquireSharedStringBuffers(const BaseVector* source);
 
+  // Acquire ownership for any string buffer that appears in source or any
```

Comment:
This is a bit confusing, it mentions that it recursively goes down layers, however the implementation only works for a single layered dictionary. Should we update it to recursively go down dictionary and constant layers? curious if there is a use case for this nesting at all?

Reply:
```
  const BaseVector* wrappedVector() const override {
    return dictionaryValues_->wrappedVector();
  }
``` 
Wrapped vector will go all the way down to the base. 
The dictionary vectors do not have string buffers in them other than the  lowest vector in the chain

Reply:
ah I see, thanks for the explanation.